File: src/main/java/seremis/geninfusion/core/proxy/CommonProxy.java
package seremis.geninfusion.core.proxy;

import cpw.mods.fml.common.FMLCommonHandler;
import cpw.mods.fml.common.SidedProxy;
import net.minecraft.entity.Entity;
import net.minecraft.world.World;
import seremis.geninfusion.handler.ServerTickHandler;
import seremis.geninfusion.lib.Localizations;

public class CommonProxy {

    @SidedProxy(clientSide = Localizations.LOC_CLIENTPROXY, serverSide = Localizations.LOC_COMMONPROXY)
    public static CommonProxy instance;

    public void registerRendering() {}// client only

    public void removeEntity(Entity entity) {
        entity.worldObj.removeEntity(entity);
    }

    public boolean isRenderWorld(World world) {
        return world.isRemote;
    }

    public boolean isServerWorld(World world) {
        return !world.isRemote;
    }

    public String playerName() {
        return "";
    }

    public void registerHandlers() {
        FMLCommonHandler.instance().bus().register(new ServerTickHandler());
    }
}

File: src/main/java/seremis/geninfusion/core/proxy/CommonProxy.scala
package seremis.geninfusion.core.proxy

import net.minecraft.entity.Entity
import net.minecraft.world.World

class CommonProxy {

    def registerRendering() {}

    def removeEntity(entity: Entity) = entity.worldObj.removeEntity(entity)

    def isRenderWorld(world: World): Boolean = world.isRemote

    def  isServerWorld(world: World): Boolean = !world.isRemote

    def playerName(): String = ""

    def registerHandlers() {}
}
